spec/controllers/sessions_controller_spec.rb

```diff
@@ -39,17 +39,112 @@
       before do
         @user = create(:user)
         @user.update_attribute :remember_token, "old-token"
-        post :create, session: { email: @user.email, password: @user.password }
       end
 
-      it { should redirect_to_url_after_create }
+      context "without configuration" do
+        before do
+          post :create, session: { email: @user.email, password: @user.password }
+        end
+
+        it { should redirect_to_url_after_create }
+
+        it "sets the user in the clearance session" do
+          expect(controller.current_user).to eq @user
+        end
 
-      it "sets the user in the clearance session" do
-        expect(controller.current_user).to eq @user
+        it "should not change the remember token" do
+          expect(@user.reload.remember_token).to eq "old-token"
+        end
       end
 
-      it "should not change the remember token" do
-        expect(@user.reload.remember_token).to eq "old-token"
+      context "with configuration" do
+        context "with a custom cookie name" do
+          before do
+            with_custom_config(:cookie_name, 'custom-cookie'){
+              post :create, session: { email: @user.email, password: @user.password }
+            }
+          end
+
+          it "sets a custom cookie name in the header" do
+            expect(response.headers['Set-Cookie']).to match /custom-cookie=old-token/
+          end
+        end
+
+        context 'with cookie path set' do
+          before do
+            with_custom_config(:cookie_path, '/cool-page'){
+              post :create, session: { email: @user.email, password: @user.password }
+            }
+          end
+          it 'sets a path cookie' do
+            expect(response.headers['Set-Cookie']).to match /path=\/cool-page/
+          end
+        end
+
+        context 'if httponly is set' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.